app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt

```diff
@@ -88,8 +83,8 @@ fun SignInScreen(
           }
 
   // If the user is already signed in, navigate to the map screen
-  LaunchedEffect(currUser) {
-    if (currUser != null) {
+  LaunchedEffect(authViewModel.isUserLoggedIn()) {
+    if (authViewModel.isUserLoggedIn()) {
```

Comment:
Praise: Great, using only the VMs currUser variable is soo much cleaner, thanks for the fix!